Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -380,6 +389,17 @@ ResourcePath ExtractLocalPathFromResourceName(
   return resource_name.PopFirst(5);
 }
 
+std::unique_ptr<MaybeDocument> InvalidMaybeDocument() {
```

Comment:
Can you add a comment briefly explaining why it's preferable to return a valid empty object instead of a null pointer?